src/basic/build.h

```diff
@@ -1,6 +1,11 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
+#include "macro.h"
+#include "version.h"
+
 extern const char* const systemd_features;
 
+#define PROJECT_VERSION_STR STRINGIFY(PROJECT_VERSION)
```

Comment:
This can be defined through conf.set_quoted in meson (also udev-ctrl.c and udevadm.h can be ported over)